contracts/storage/ShardingTableStorage.sol

```diff
@@ -37,6 +39,27 @@ contract ShardingTableStorage is INamed, IVersioned, HubDependent {
         nodesCount--;
     }
 
+    function getStakeWeightedAverageAsk() external view returns (uint256) {
+        return totalStake > 0 ? weightedAskSum / totalStake : 0;
+    }
+
+    function setTotalStake(uint96 _totalStake) external onlyContracts {
+        totalStake = _totalStake;
+    }
+
+    function setWeightedAskSum(uint256 _weightedAskSum) external onlyContracts {
+        weightedAskSum = _weightedAskSum;
+    }
+
+    function onStakeChanged(uint96 oldStake, uint96 newStake, uint96 ask) external onlyContracts {
+        weightedAskSum = weightedAskSum - (ask * oldStake) + (ask * newStake);
```

Comment:
I would check if there is a way to get underflow here

Reply:
`(ask * oldStake)` is always included in `weightedAskSum`